### ISSUE TITLE
GNOME Shell extension windowNavigator breaks "Activities" overview in GNOME 40

### ISSUE THREAD

**Issue by das-g** [NixOS/nixpkgs#122817]
**Describe the bug**
When the GNOME Shell extension [windowNavigator](https://extensions.gnome.org/extension/10/windownavigator/) is installed (which it seems to be by default<sup>1</sup>) and enabled, then the GNOME 40 "Activities" overview view doesn't work as expected:

- It doesn't show an overview of the windows
- It doesn't show the frame around the window overview indicating the current workspace
- It can't be closed (return no normal view) by hovering or clicking "Activities" at the top left
- It can't be closed (return no normal view) by hitting the <kbd>Win</kbd> key
- It can't be closed (return no normal view) by hitting the <kbd>Esc</kbd> key
- Typing text doesn't write into the search input field and doesn't list matching applications
  - Also not in the list-of-available-applications view (see below how to access it)
- Clicking application icons in the launch bar at the bottom doesn't seem to have any effect
- Clicking the 9-dot icon in the launch bar doesn't seem to have any effect (should change to view showing the list of available applications)

What still works:
- It **can** be closed (return no normal view) by 3-finger down-swipe on the touch pad
- The list of available applications **can** be accessed by 3-finger up-swipe on the touch pad
  - From that one, one **can** return to the normal view by hitting the <kbd>Win</kbd> key
  - From that one, one **can** return to the normal view by hitting the <kbd>Esc</kbd> key _twice_, even though that seems to go through the activities overview view again
  - Applications can be launched here, which also returns one to the normal view
- The "Applications" and "Places" menus provided by extensions [Applications Menu](https://extensions.gnome.org/extension/6/applications-menu/) and [Places Status Indicator](https://extensions.gnome.org/extension/8/places-status-indicator/) if enabled (also installed by default<sup>1</sup>)
- Running commands or launching applications through <kbd>Alt</kbd>+<kbd>F2</kbd>

<sup>1</sup>It's listed as a "System extension" on https://extensions.gnome.org/local/, but I didn't list it in my `/etc/nixos/configuration.nix`.

**To Reproduce**
1. Have NixOS on channel `nixos-unstable` with GNOME
   In `configuration.nix`:
   ```nix
   { config, lib, pkgs, ... }:
   {
      # ...
      services.xserver.desktopManager.gnome.enable = true;
      # ...
   }
   ```
2. Enable GNOME Shell extension "windowNavigator"
    e.g. through https://extensions.gnome.org/local/ in Firefox, which requires:
    - In `configuration.nix`:
      ```nix
      { config, lib, pkgs, ... }:
      {
         # ...
         services.gnome.chrome-gnome-shell.enable = true;
         nixpkgs.config.firefox.enableGnomeExtensions = true;
         # ...
      }
      ```
    - Firefox extension https://wiki.gnome.org/Projects/GnomeShellIntegrationForChrome
3. Log into GNOME
4. Hit <kbd>Win</kbd> key

**Expected behavior**
- **Shows** an overview of the windows
- **Shows** a the frame around the window overview indicating the current workspace
- Overview **can** be closed (return no normal view) by hovering or clicking "Activities" at the top left
- Overview **can** be closed (return no normal view) by hitting the <kbd>Win</kbd> key
- Overview **can** be closed (return no normal view) by hitting the <kbd>Esc</kbd> key
- Typing text **does** write into the search input field which causes matching applications to be listed
  - Also in the list-of-available-applications view
- Clicking application icons in the launch bar at the bottom **launches** said application
- Clicking the 9-dot icon in the launch bar changes to the view showing the list of available applications

**Screenshots**
actual (with windowNavigator enabled):
![Bildschirmfoto von 2021-05-13 12-41-23](https://user-images.githubusercontent.com/97746/118114991-a069b980-b3e8-11eb-9d54-4947043812af.png)

expected (and actual with windowNavigator disabled):
![Bildschirmfoto von 2021-05-13 12-41-06](https://user-images.githubusercontent.com/97746/118115057-b9726a80-b3e8-11eb-8c47-88ee52b076bd.png)

**Additional context**
This problem didn't occur before GNOME 40

**Notify maintainers**

(I can't find the "windowNavigator" extension in nixpkgs. Is GNOME by default shipped with that extension included already by upstream?)

- @NixOS/gnome

**Metadata**
Result of `nix-shell -p nix-info --run "nix-info -m"`:
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.35, NixOS, 21.05pre288911.65d6153aec8 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixos-21.05pre288911.65d6153aec8, nixos-unstable-21.03pre257780.e9158eca70a"`
 - channels(das-g): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- "gnome3.gnome-shell"
- "gnome3.gnome-shell-extensions"
# a list of nixos modules affected by the problem
module:
- "nixos/modules/services/x11/desktop-managers/gnome.nix"
```

**Comment by jtojnar** [comment]
It is part of `gnome.gnome-shell-extensions package`.

Upstream issue: https://gitlab.gnome.org/GNOME/gnome-shell-extensions/-/issues/307

**Comment by jtojnar** [comment]
Gonig to be fixed in #122997.
